obspy/core/tests/test_event.py

```diff
@@ -681,30 +717,22 @@ def test_same_resource_id_different_referred_object(self):
         """
         Tests the handling of the case that different ResourceIdentifier
         instances are created that have the same resource id but different
-        objects. This should not happen and thus a warning should be emitted.
+        objects. The referred objects should still return the same objects
+        used in the ResourceIdentifier construction or set_referred_object
+        call
         """
         object_a = UTCDateTime(1000)
         object_b = UTCDateTime(1001)
-        self.assertEqual(object_a is object_b, False)
+        self.assertFalse(object_a is object_b)
         id = 'obspy.org/tests/test_resource'
-        res_a = ResourceIdentifier(id=id,
-                                   referred_object=object_a)
+        res_a = ResourceIdentifier(id=id, referred_object=object_a)
         # Now create a new resource with the same id but a different object.
-        # This will raise a warning.
-        with warnings.catch_warnings(record=True):
-            warnings.simplefilter('error', UserWarning)
-            self.assertRaises(UserWarning, ResourceIdentifier,
-                              id=id,
-                              referred_object=object_b)
-            # Now ignore the warning and actually create the new
-            # ResourceIdentifier.
-            warnings.simplefilter('ignore', UserWarning)
-            res_b = ResourceIdentifier(id=id,
-                                       referred_object=object_b)
-        # Object b was the last to added, thus all resource identifiers will
-        # now point to it.
-        self.assertEqual(object_b is res_a.get_referred_object(), True)
-        self.assertEqual(object_b is res_b.get_referred_object(), True)
+        # This will no longer raise a warning
+        res_b = ResourceIdentifier(id=id, referred_object=object_b)
```

Comment:
I'm fine with the behavior of the code but I think it should really still raise a warning if two non-equal objects share the same resource identifiers. The whole purpose of resource ids is to point to unique or at least equal objects. If that is not the case some logic is wrong and the users should be notified.